### ISSUE TITLE
can't compile FUNWAVE-TVD (version 3) -- ambigous variable

### ISSUE THREAD

**Issue by gsalgado213** [fengyanshi/FUNWAVE-TVD#3]
@fengyanshi (Matt and I are working on installing the latest cloned version of FUNWAVE on my new machine. When compiling, I'm getting the following error. It complains that MPI_MAX function/action is defined in module param, but it is not. I've checked all the files *.F with grep and there is no instance of it anywhere, except for MPI_ALLREDUCE function signature in init.F, misc.F, and statistics.F only. Any idea what's going on? Thanks, Gabriela.

`gabriela@malej-linux:~/FUNWAVE-TVD$ make
/usr/bin/cpp  -P -traditional  -P -traditional  -DDOUBLE_PRECISION  -DPARALLEL -DSAMPLES -DCARTESIAN                     mod_param.F > mod_param.f90
mpif90  -c     mod_param.f90
/usr/bin/cpp  -P -traditional  -P -traditional  -DDOUBLE_PRECISION  -DPARALLEL -DSAMPLES -DCARTESIAN                     mod_global.F > mod_global.f90
mpif90  -c     mod_global.f90
/usr/bin/cpp  -P -traditional  -P -traditional  -DDOUBLE_PRECISION  -DPARALLEL -DSAMPLES -DCARTESIAN                     mod_input.F > mod_input.f90
mpif90  -c     mod_input.f90
/usr/bin/cpp  -P -traditional  -P -traditional  -DDOUBLE_PRECISION  -DPARALLEL -DSAMPLES -DCARTESIAN                     main.F > main.f90
mpif90  -c     main.f90
/usr/bin/cpp  -P -traditional  -P -traditional  -DDOUBLE_PRECISION  -DPARALLEL -DSAMPLES -DCARTESIAN                     bc.F > bc.f90
mpif90  -c     bc.f90
/usr/bin/cpp  -P -traditional  -P -traditional  -DDOUBLE_PRECISION  -DPARALLEL -DSAMPLES -DCARTESIAN                     fluxes.F > fluxes.f90
mpif90  -c     fluxes.f90
/usr/bin/cpp  -P -traditional  -P -traditional  -DDOUBLE_PRECISION  -DPARALLEL -DSAMPLES -DCARTESIAN                     init.F > init.f90
mpif90  -c     init.f90
init.f90:501.62:

      CALL MPI_ALLREDUCE(EtaBlowVal,myvar_tmp,1,MPI_SP,MPI_MAX,MPI_COMM_WORLD,i
                                                              1
Error: Name 'mpi_max' at (1) is an ambiguous reference to 'mpi_max' from module 'param'
make: *** [init.o] Error 1
`

**Comment by fengyanshi** [comment]
It is a strange problem I’ve never seen. MPI_MAX is defined internally in the MPI module but it complaint about using it. Can you check if the old version (2.0) works? Let me know.

**Comment by gsalgado213** [comment]
I've verified with funwave 2.1 and it also shows the same problem.

**Comment by fengyanshi** [comment]
I guess something wrong with the mpi lib.

**Comment by gsalgado213** [comment]
It seems like it. We will reboot the computer and try again. I'll let you know the outcome.
Thank you.

**Comment by malej** [comment]
‎@gsalgado213 - we will need to rebuild your MPI compilers on Friday  to fix this..                                                                                                                                                                                                                From: gsalgado213Sent: Thursday, June 23, 2016 7:50 AMTo: fengyanshi/FUNWAVE-TVDReply To: fengyanshi/FUNWAVE-TVDSubject: Re: [fengyanshi/FUNWAVE-TVD] can't compile FUNWAVE-TVD (version 3) -- ambigous variable (#3)It seems like it. We will reboot the computer and try again. I'll let you know the outcome.
Thank you. 

—You are receiving this because you are subscribed to this thread.Reply to this email directly, view it on GitHub, or mute the thread.

**Comment by gsalgado213** [comment]
We took care of the problem. It was generated by the old version of our MPI compiler that did not sit well with `use mpi` directive and we were forced to switch to `include mpi.h`. As such, we had to at the time change the location of `implicit none` for that version of the MPI. When we upgraded to the new Ubuntu and newest MPI compilers and went back to `use mpi` original directive, everything compiled and linked fine. Closing the issue.
